libraries/AP_UAVCAN/AP_UAVCAN_Servers.cpp

```diff
@@ -83,6 +83,29 @@ class AP_UAVCAN_FileEventTracer : public uavcan::dynamic_node_id_server::IEventT
     }
 };
 
+
+class AP_UAVCAN_RestartRequestHandler : public uavcan::IRestartRequestHandler {
+public:
+    bool handleRestartRequest(uavcan::NodeID request_source) override {
+        // swiped from reboot handling in GCS_Common.cpp
+        AP_Notify *notify = AP_Notify::instance();
+        if (notify) {
+            AP_Notify::flags.firmware_update = 1;
+            notify->update();
+        }
+        // force safety on
+        hal.rcout->force_safety_on();
```

Comment:
At somepoint maybe we should be handling this inside of the scheduler reboot call? Seem to be having to copy paste this around, and having consistent behavior on this is very valuable actually.

Reply:
I very much agree, and I looked at where to factor that code to.

Inside the scheduler felt wrong.

AFAICS we don't really have a singleton floating around representing the board itself.

Perhaps Just `hal.reboot()` or `hal.util->reboot()`?

Reply:
I'm fine with either of those. The hooking of AP_Param flush from the HAL is a bit weird though. Ideally should be done for consistency, but only the `rcout` stuff is really reasonable to move into the HAL I think. You could just add a `AP::reboot()` I guess which could then do all the above, but it's a bit weird?